qep-403-clientside-comments.md

```diff
@@ -0,0 +1,126 @@
+# QGIS Enhancement: Title
+
+**Date** 2025/11/24
+
+**Author** Dave Signer (@signedav)
+
+**Contact** david at opengis dot ch
+
+**Version** QGIS 4.0 or 4.2
+
+# Summary
+
+This QEP introduces the possibility to change field comments on the client side, which are stored in the QGIS configuration. The solution does **not propose** to introduce more possibilities to change the server-side comments. It's about **client-side comments** that override the server-side comments in QGIS.
+
+The need for this is because server-side comments have their limitations:
+- No user rights to change them
+- When you change them, they are changed for everyone connecting to the database (you could potentially break other QGIS project workflows).
+
+That's why client-side comments should be introduced.
+
+## The existing server-side comments
+
+More context regarding the server-side comments.
+
+### Where are they defined?
+
+Column comments are defined directly in the database. They are mainly used in `PostgreSQL`, for example:
+
+```
+COMMENT ON COLUMN plan.chcantoncode.ilicode IS 'This is the abbreviation of the canton (e.g. ZH for Zurich)';
+```
+
+Another provider that utilizes them is the `GrassLayers`.
+
+They are read on `loadFields` within the provider and are then stored in the `QgsField` objects.
+
+### Where are they used in QGIS?
+
+The comments are displayed as hover tooltips in the forms.
+
+![current tooltip](images/qep403/current-tooltip.png)
+
+### Where can these comments be changed?
+
+In the browser panel you can *edit* / *add* such comments to columns.
+
+![browser](images/qep403/browser.png)
+
+Also in the layer properties "Fields" section you can *add* a comment on adding a *new* column. However, you *cannot edit* existing ones.
+
+![current fields section](images/qep403/fields.png)
+
+You can *view* these comments in the layer properties "Attribute Form" section, but you *cannot edit* them.
+
+![current attribute form section](images/qep403/current-attributeform-config.png)
+
+## Proposed Solution
+
+Introduce **client-side comments** stored in the QGIS configuration (like the aliases), which will override the server-side comments. This means if a client-side comment exists, it will be used. Otherwise, the server-side comment will be used.
+
+### Comment definition
+
+In the `QgsAttributeTypeDialog` the comment is editable (`QLineEdit`). It offers the same options as the alias, including the possibility of being data-defined by expressions, utilizing a `QgsPropertyOverrideButton`.
+
+![server default](images/qep403/default-server-side.png)
+
+When we edit the comment in the attribute form settings, a client-side comment is defined and will be taken into account.
+
+![client set](images/qep403/client-side-comment.png)
+
+If no client-side comment is defined (it's `NULL`- removed again using the x-arrow on the right), the server-side comment is taken into account again.
```

Comment:
I think clearing (i.e. nulling) the comment should not result in "NULL" being displayed in this widget, as that would be quite user-unfriendly. Is it possible to show the server-side comment instead of NULL and display it in a different style? (ie different color/italic/something)